### DISCUSSION TITLE
Shader constant buffers (uniforms) members reflection

### DISCUSSION THREAD

**Discussion by cosator** [DiligentGraphics/DiligentEngine#234] [General]
Hi,
I'm working on adding Diligent as an option for graphics backends for U3D game engine (former Urho3D game engine). The intention is to fully replace in the future the existing DirectX11 and OpenGL backends with Diligent. The current implementation of U3D game engine requires reflection of member data of shader constant buffers (uniforms). I'm already adding that capability to Diligent and I'm interested in getting this change to mainline. I think I can also make the capability optional with a compile time switch. Please let me know if there is interest into accepting a pull request for this feature.
Thank you so much!

**Comment by TheMostDiligent** [comment]
Hello!
Shader constant buffer reflection has recently been added to the API - please take a look at [IShader::GetConstantBufferDesc](https://github.com/DiligentGraphics/DiligentCore/blob/master/Graphics/GraphicsEngine/interface/Shader.h#L863)

**Reply by cosator** [reply]
Awesome! If I knew that yesterday, I would have saved me a few hours of prototyping. :)
Member sizes would have been a nice to have since they come for free in directx and vulkan, but would have required calculation on opengl (I know that they can be calculated from the info in ShaderCodeVariableDesc).
By the way, great job on this library!
